src/rust/engine/options/src/config.rs

```diff
@@ -336,21 +338,46 @@ impl ConfigReader {
         id.name("_", NameTransform::None)
     }
 
-    fn get_value(&self, id: &OptionId) -> Option<&Value> {
+    fn get_from_section(&self, section_name: &str, option_name: &str) -> Option<&Value> {
         self.config
             .value
-            .get(id.scope.name())
-            .and_then(|table| table.get(Self::option_name(id)))
+            .get(section_name)
+            .and_then(|table| table.get(option_name))
+    }
+
+    fn get_value(&self, id: &OptionId) -> Option<&Value> {
+        let option_name = Self::option_name(id);
+        self.get_from_section(id.scope.name(), &option_name)
+            .or(self.get_from_section(DEFAULT_SECTION, &option_name))
     }
 
     fn get_list<T: FromValue + Parseable>(
```

Comment:
Two questions:

1. can you confirm that the Python parser gives `foo.opt == [1, 2]` for this config?
  
    ```toml
    [DEFAULT]
    opt = [1]
    
    [foo]
    opt.add = [2]
    ```

2. this PR doesn't seem to change apply the same change to dict-value options, is that intentional? The Python parser doesn't use `DEFAULT` for them?

Reply:
1. Yep. This falls out of the fact that in the Python implementation "config default" is treated as a rank. I'll add to the python tests to emphasize this.
2. Ooops, good catch. Will fix (and add tests on the Python side too, to for emphasis).